tests/integration/solidity_contracts/PlainOpcodes/PlainOpcodes.t.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity 0.8.10;
+
+import "forge-std/Test.sol";
+import {PlainOpcodes} from "./PlainOpcodes.sol";
+import {Counter} from "./Counter.sol";
+
+contract PlainOpcodesTest is Test {
+    PlainOpcodes plainOpcodes;
+    Counter counter;
+
+    function setUp() public {
+        counter = new Counter();
+        plainOpcodes = new PlainOpcodes(address(counter));
+    }
+
+    function testOpcodeExtCodeCopyReturnsCounterCode(
```

Comment:
having a couple high level comments around the assembly blocks explaining what the outcome is would help with readability, imo.